php/commands/core.php

```diff
@@ -1150,6 +1165,40 @@ private function get_updates( $assoc_args ) {
 		return array_values( $updates );
 	}
 
+	private function remove_unused_files( $version_from, $version_to, $locale ) {
+		if ( ! $version_from || ! $version_to ) {
+			WP_CLI::warning( 'Failed to find WordPress version. Please cleanup unused files manually.' );
+			return;
+		}
+
+		$old_checksums = self::get_core_checksums( $version_from, $locale ? $locale : 'en_US' );
+		$new_checksums = self::get_core_checksums( $version_to, $locale ? $locale : 'en_US' );
+
+		if ( empty( $old_checksums ) || empty( $new_checksums ) ) {
+			WP_CLI::warning( 'Failed to fetch checksums. Please cleanup unused files manually.' );
+			return;
+		}
+
+		$files_to_remove = array_diff( array_keys( $old_checksums ), array_keys( $new_checksums ) );
+
+		if ( ! empty( $files_to_remove ) ) {
+			WP_CLI::log( 'Cleaning up unused files...' );
+
+			$count = 0;
+			foreach ( $files_to_remove as $file ) {
+				if ( file_exists( ABSPATH . $file ) ) {
+					unlink( ABSPATH . $file );
+					WP_CLI::log( 'File removed: ' . $file );
+					$count++;
+				}
+			}
+
+			if ( $count ) {
+				WP_CLI::success( number_format( $count ) . ' unused files cleaned up' );
+			}
```

Comment:
If `$count` ended up being empty, can we have a message that says there weren't any files to be removed.